### ISSUE TITLE
Missing metric plots

### ISSUE THREAD

**Issue by SimonePietrogiacomi** [ultralytics/yolov5#11078]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Validation

### Bug

Most of the times my `best.pt` don't produce any metrics plot in validation. It finish its execution without any error but in the "runs/val/exp/" directory there are only batch labels and preds, not a single plot. The only things I've changed from the original updated repo are the hyperparameters, the fitness weights and I've added the custom yaml dataset file. Sometimes it shows the warning `WARNING ⚠️ ConfusionMatrix plot failure: In set_text: Could not load glyph (error code 0x14)` and create some of the files it should create, but they are empty.

### Minimal Reproducible Example

Just launch a validation of a custom dataset yolov5s from a jupyter notebook.

`!python3 val.py --task val --data data/custom.yaml --weights runs/train/exp2/weights/best.pt --img 512`

### Additional

I don't know if it can be useful in some way, but I've trained a Yolov5s on a custom dataset with "--epochs 150". The train finished at epoch 100 (probably overfit) and if I try to resume it says that it reached the epoch 150. It's weird, but it should save the plot anyways.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@SimonePietrogiacomi plots are in Try Except clauses so that if they produce an error they don't interrupt the rest of trainings. From your error message the fonts you are using for your classes, or perhaps matplotlib or your environment are causing issues. I'd train in our Docker image for best environment compatibility if you are having these issues and your fonts are standard.

**Comment by SimonePietrogiacomi** [comment]
Thank you for your reply. I'll try in another env or, better, on another machine (I don't think I have privileges to install docker on this one)

**Comment by SimonePietrogiacomi** [comment]
I'm in Colab right now, seems like it's working. You were right, it was missing the fonts download but I don't know if there were other silent stuff missing. Ty again :)

**Comment by glenn-jocher** [comment]
You're welcome, @SimonePietrogiacomi! It's great to hear that it's working in Colab. If you encounter any further issues, feel free to reach out. Stand by if you need any assistance. 🚀
